core/vm/ovm_state_manager.go

```diff
@@ -131,7 +131,26 @@ func putContractStorage(evm *EVM, contract *Contract, args map[string]interface{
 		return nil, errors.New("Could not parse value arg in putContractStorage")
 	}
 	val := toHash(_value)
-	evm.StateDB.SetState(address, key, val)
+
+	// save the block number and address with modified key if it's not an eth_call
+	if evm.Context.EthCallSender == nil {
+		// save the value before
+		before := evm.StateDB.GetState(address, key)
+		evm.StateDB.SetState(address, key, val)
+		err := evm.StateDB.SetDiffKey(
+			evm.Context.BlockNumber,
+			address,
+			key,
+			before != val,
+		)
+		if err != nil {
+			log.Error("error", err)
```

Comment:
This will log an additional error since there are an even number or args, if you could make it be something like:
```golang
log.Error("Cannot set diff key", "msg", err)
```

Then it will log something like:

```
Error: Cannot set diff key            msg="error message"
```